__tests__/reusable.test.js

```diff
@@ -0,0 +1,162 @@
+import {useState} from 'react'
+import {renderHook, act} from 'react-hooks-testing-library'
+import {getStore, createStore, replaceStore} from '../src/reusable';
+import {reusable, ReusableProvider} from '../src/react-reusable';
+
+let store;
+let unit;
+const value = {title: 'Yo'};
+const expectToBeAStore = (object) => {
+  expect(object.units).toBeDefined();
+}
+const expectToBeAUnit = (object) => {
+  expect(object.cachedValue).toBeDefined();
+}
+describe('public store API', () => {
+  beforeEach(() => {
+    store = getStore();
+  });
+  it('should allow to get a store', () => {
+    expectToBeAStore(store);
+  });
+  it('should allow to create a store', () => {
+    expectToBeAStore(createStore());
+  });
+  it('should allow to replace the store', () => {
+    const newStore = createStore();
+    replaceStore(newStore);
+    expect(getStore()).toBe(newStore);
+  });
+  it('should allow to create a unit', () => {
+    const fn = () => {};
+    store.createUnit(fn);
+    expectToBeAUnit(store.getUnit(fn));
+  });
+  it('should allow to subscribe to creating a unit', () => {
+    const fn = () => {};
+    let result;
+    store.onUnitsChanged(() => {
+      result = store.getUnit(fn);
+    });
+    store.createUnit(fn);
+    expectToBeAUnit(result);
+  });
+  it('should not allow to get a unit without creating it', () => {
+    const fn = () => {};
+
+    expect(() => store.getUnit(fn)).toThrow();
+  });
+  it('should not allow to create a unit twice', () => {
+    const fn = () => {};
+    store.createUnit(fn);
+    expect(() => store.createUnit(fn));
+  });
+  it('should allow to unsubscribe from subscribing to creating a unit', () => {
+    expect.assertions(0);
+    const fn = () => {};
+    const unsubscribe = store.onUnitsChanged(() => {
+      expect(true).toBe(false); // should not reach here
+    });
+    unsubscribe();
+    store.createUnit(fn);
+  });
+
+});
+describe('public unit API', () => {
+  beforeEach(() => {
+    const fn = () => value;
+    store.createUnit(fn);
+    unit = store.getUnit(fn);
+  });
+
+  it('should allow to get the cached value', () => {
+    expect(unit.cachedValue).toBeDefined();
+  });
+
+  it('should allow to run the unit', () => {
+    expect(unit.run()).toBe(value);
+  });
+
+  it('should cache the unit value', () => {
+    unit.run();
+    expect(unit.cachedValue).toBe(value);
+  });
+  it('should allow to subscribe to value change', () => {
+    let result;
+    unit.subscribe((cachedValue) => {
+      result = cachedValue;
+    });
+    unit.run();
+    unit.notify();
+    expect(result).toBe(value);
```

Comment:
instead of result you can pass a spy and then expect(spy).toHaveBeenCalledWith(value)